sdk/ruby/lib/chain/connection.rb

```diff
@@ -186,6 +186,15 @@ def http
       if @url.scheme == 'https'
         @http.use_ssl = true
         @http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+        if @opts.key?(:root_ca_certs_path)
+          @http.ca_file = @opts[:root_ca_certs_path]
+        end
+        if @opts.key?(:client_cert_path) && opts.key?(:client_key_path)
```

Comment:
Nit: I have the feeling that unless we do more robust sanity checks here (i.e. raise an exception if one of the cert or key is missing), it may be safe to check for and set each of these in their own `if` blocks.